nanoCAT/recipes/coordination_number.py

```diff
@@ -0,0 +1,129 @@
+import numpy as np
+from itertools import combinations
+from typing import Dict, Tuple, List
+from scm.plams import Molecule
+from FOX import group_by_values
+from nanoCAT.bde.guess_core_dist import guess_core_core_dist
+
+#: A nested dictonary
+NestedDict = Dict[str, Dict[int, List[int]]]
+
+
+def get_indices(mol: Molecule) -> Dict[str, np.ndarray]:
+    """Construct a dictionary with atomic symbols as keys and arrays of indices as values."""
+    elements = [at.symbol for at in mol.atoms]
+    symbol_enumerator = enumerate(elements)
+    idx_dict = group_by_values(symbol_enumerator)
+    for k, v in idx_dict.items():
+        idx_dict[k] = np.fromiter(v, dtype=int, count=len(v))
+    return idx_dict
+
+
+def idx_pairs(idx_dict: Dict[str, np.ndarray]) -> Tuple[np.ndarray, np.ndarray]:
+    """Construct two arrays of indice-pairs of all possible combinations in idx_dict.
+
+    The combinations, by definition, do not contain any atom pairs where ``at1.symbol == at2.symbol``
```

Comment:
[Travis](https://travis-ci.org/github/nlesc-nano/nano-CAT/jobs/667313963#L847-L854) is also complaining that this line is too long.
As this is in a docstring it is, in my opinion, not as important to stay under the 100 character limit.
You can either move part of the sentence to the next line or you can add ``# noqa`` add the end of the docstring (after the ``"""``)